pilot/pkg/networking/core/v1alpha3/cluster_builder_test.go

```diff
@@ -2673,6 +2673,49 @@ func TestBuildUpstreamClusterTLSContext(t *testing.T) {
 				err: nil,
 			},
 		},
+		{
+			name: "tls mode SIMPLE, with EcdhCurves specified in Mesh Config",
```

Comment:
Can you please also add a test case for MUTUAL?

Reply:
Added additional test cases

Reply:
Sorry I meant to ask ISTIO_MUTUAL as well. If Mesh config has ecdh curves configured, for Istio Mutual we should not add. Can you please add that test also? Here and below?

Reply:
Done